scripts/bfasst/compare_waveforms/file_parsing/parse_files.py

```diff
@@ -58,19 +73,21 @@ def parse_multiple(file, reversed_file):
         if not_port is False:
             for port in i.ports:
                 if str(port.direction) == "Direction.OUT":
-                    data["output_list"].append(port.name)
-                    data["total_list"].append(port.name)
-                    data["output_bits_list"].append(len(port.pins) - 1)
+                    output_list.append(port.name)
+                    total_list.append(port.name)
+                    output_bits_list.append(len(port.pins) - 1)
                 elif str(port.direction) == "Direction.IN":
-                    data["input_list"].append(port.name)
-                    data["total_list"].append(port.name)
-                    data["input_bits_list"].append(len(port.pins) - 1)
-
-            if "clk" in data["input_list"]:
-                if data["input_list"].index("clk") == 0:
-                    data["input_list"].append(data["input_list"].pop(0))
+                    input_list.append(port.name)
+                    total_list.append(port.name)
+                    input_bits_list.append(len(port.pins) - 1)
+            if "clk" in input_list:
+                if input_list.index("clk") == 0:
+                    input_list.append(input_list.pop(0))
         else:
             not_port = False
+    data = set_data(
+        input_list, output_list, input_bits_list, output_bits_list, total_list
+    )
```

Comment:
Do you know you are going to have data in all sub_elements of `data`, ie input_list, output_list, input_bits_list, output_bits_list, total_list?  If that's the case, then setdefault won't do anything helpful.  Simply setting up `data["input"] = []` before accessing it is probably faster and cleaner.

The reason you would use setdefault is if you want to make sure you don't overwrite anything, but if nothing is there, you want to make the value for that key a list (or string, or number, etc.).  If you may or may not have a dictionary item initialized at the end of the function, then it makes sense to use setdefault when you need to, because that preserves the non-initialized state if you never use it, but gives you a deterministic default when you do use it.  That means that you never have empty lists in a dictionary.  If the key exists in the dictionary, it's value is guaranteed to be non-empty, since it's only empty as a default right before you change it.  Does that make sense?